pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -851,6 +855,34 @@ ${compileResult.stdout}
         ? BuildOutput(hookOutputJson)
         : LinkOutput(hookOutputJson);
   }
+
+  /// Reads the user-defines from a pubspec.yaml in the suggested location.
+  ///
+  /// SDKs do not have to follow this, they might support user-defines in a
+  /// different way.
+  ///
+  /// The [pubspec] is expected to be the decoded yaml, a Map.
+  static Map<String, Map<String, Object?>> readHooksUserDefinesFromPubspec(
+    Map<Object?, Object?> pubspec,
+  ) {
+    final hooks = pubspec['hooks'];
+    if (hooks is! Map) {
+      return {};
+    }
+    final userDefines = hooks['user_defines'];
+    if (userDefines is! Map) {
+      return {};
+    }
+    return {
+      for (final MapEntry(:key, :value) in userDefines.entries)
```

Comment:
This parses only one level deep - this information should be transparent in the `input.userDefines` the user sees, where there currently is an `Object?`.